test/websocket-server.test.js

```diff
@@ -510,6 +510,41 @@ describe('WebSocketServer', () => {
         });
       });
     });
+
+    it('uses the parent `WebSocket` class', (done) => {
+      const custom_field = '123';
+
+      class CustomWebSocket extends WebSocket.WebSocket {
+        get custom_field() {
+          return custom_field;
+        }
+      }
+
+      const server = http.createServer();
+
+      server.listen(0, () => {
+        const wss = new WebSocket.Server({
+          noServer: true,
+          WebSocket: CustomWebSocket
+        });
+
+        server.on('upgrade', (req, socket, head) => {
+          wss.handleUpgrade(req, socket, head, (ws) => {
+            assert.ok(ws instanceof CustomWebSocket);
+            assert.deepStrictEqual(ws.custom_field, custom_field);
+            ws.close();
+          });
+        });
+
+        const ws = new WebSocket(`ws://localhost:${server.address().port}`);
+
+        ws.on('open', () => {
+          ws.on('close', () => {
+            server.close(done);
+          });
+        });
+      });
+    });
```

Comment:
```suggestion
    it('honors the `WebSocket` option', (done) => {
      class CustomWebSocket extends WebSocket.WebSocket {
        get foo() {
          return 'foo';
        }
      }

      const wss = new WebSocket.Server(
        {
          port: 0,
          WebSocket: CustomWebSocket
        },
        () => {
          const ws = new WebSocket(`ws://localhost:${wss.address().port}`);

          ws.on('open', ws.close);
        }
      );

      wss.on('connection', (ws) => {
        assert.ok(ws instanceof CustomWebSocket);
        assert.strictEqual(ws.foo, 'foo');
        wss.close(done);
      });
    });
```

Also, please move it under the "options" section.